charts/redpanda/templates/secrets.yaml

```diff
@@ -351,10 +351,14 @@ stringData:
     rpk --config "$CONFIG" redpanda config set redpanda.rack "${RACK}"
       {{- end }}
     {{- end }}
+    {{- $cloudStorageSecretKey := "cloud_storage_secret_key" }}
+    {{- if .Values.storage.tiered.credentialsSecretRef.azure }}
+      {{- $cloudStorageSecretKey = "cloud_storage_azure_shared_key" }}
+    {{- end }}
     {{- if and .Values.storage.tiered.credentialsSecretRef.name .Values.storage.tiered.credentialsSecretRef.key }}
     set +x
-    echo Setting cloud_storage_secret_key configuration
-    rpk redpanda config --config "$CONFIG" set cloud_storage_secret_key $CLOUD_STORAGE_SECRET_KEY
+    echo Setting {{ $cloudStorageSecretKey }} configuration
+    rpk redpanda config --config "$CONFIG" set {{ $cloudStorageSecretKey }} $CLOUD_STORAGE_SECRET_KEY
```

Comment:
Looking at the history of Redpanda the configuration key could be changed for any other S3 equivalent storage.

I would suggest
```
    credentialsSecretRef:
      configurationKey: cloud_storage_secret_key
      name: XYZ
      key: ABC
```

That way you would be able to change that for azure. 

BTW We have opened issue for adding access key to this script too.

Reply:
@RafalKorepta how about now?